Index: kernel5_4/drivers/net/usb/cdc_ncm.c
===================================================================
--- kernel5_4.orig/drivers/net/usb/cdc_ncm.c
+++ kernel5_4/drivers/net/usb/cdc_ncm.c
@@ -658,9 +658,9 @@ static int cdc_ncm_setup(struct usbnet *
 	/* be conservative when selecting intial buffer size to
 	 * increase the number of hosts this will work for
 	 */
-	def_rx = min_t(u32, CDC_NCM_NTB_DEF_SIZE_RX,
+	def_rx = max_t(u32, CDC_NCM_NTB_DEF_SIZE_RX,
 		       le32_to_cpu(ctx->ncm_parm.dwNtbInMaxSize));
-	def_tx = min_t(u32, CDC_NCM_NTB_DEF_SIZE_TX,
+	def_tx = max_t(u32, CDC_NCM_NTB_DEF_SIZE_TX,
 		       le32_to_cpu(ctx->ncm_parm.dwNtbOutMaxSize));
 
 	/* clamp rx_max and tx_max and inform device */
